atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/atomix/AtomixServerImplementation.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.timelock.atomix;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.palantir.atlasdb.timelock.ServerImplementation;
+import com.palantir.atlasdb.timelock.TimeLockServices;
+import com.palantir.atlasdb.timelock.config.AtomixConfiguration;
+import com.palantir.atlasdb.timelock.config.AtomixSslConfiguration;
+import com.palantir.atlasdb.timelock.config.TimeLockServerConfiguration;
+import com.palantir.lock.impl.LockServiceImpl;
+import com.palantir.remoting1.config.ssl.SslConfiguration;
+
+import io.atomix.AtomixReplica;
+import io.atomix.catalyst.transport.Address;
+import io.atomix.catalyst.transport.Transport;
+import io.atomix.catalyst.transport.netty.NettyTransport;
+import io.atomix.copycat.server.storage.Storage;
+import io.atomix.group.DistributedGroup;
+import io.atomix.group.LocalMember;
+import io.atomix.variables.DistributedLong;
+import io.atomix.variables.DistributedValue;
+
+public class AtomixServerImplementation implements ServerImplementation {
+    private static final Logger log = LoggerFactory.getLogger(AtomixServerImplementation.class);
+
+    private AtomixReplica replica;
+    private LocalMember localMember;
+
+    @Override
+    public void onStartup(TimeLockServerConfiguration configuration) {
+        AtomixConfiguration atomix = (AtomixConfiguration) configuration.algorithm();
+        replica = AtomixReplica.builder(new Address(configuration.cluster().localServer()))
+                .withStorage(Storage.builder()
+                        .withDirectory(atomix.storageDirectory())
+                        .withStorageLevel(atomix.storageLevel())
+                        .build())
+                .withTransport(createTransport(atomix.security()))
+                .build();
+        AtomixRetryer.getWithRetry(() -> replica.bootstrap(
+                configuration.cluster()
+                        .servers()
+                        .stream()
+                        .map(Address::new)
+                        .collect(Collectors.toSet())));
+
+        DistributedGroup timeLockGroup = DistributedValues.getTimeLockGroup(replica);
```

Comment:
Okay, done (though I've renamed it to `startAtomix()` which I find captures what this part does better).